corehq/apps/app_manager/views/utils.py

```diff
@@ -466,20 +467,36 @@ def handle_shadow_child_modules(app, shadow_parent):
             app.delete_module(child.unique_id)
 
     # Add new modules
-    for shadow_child in source_module_children:
+    for source_child in source_module_children:
```

Comment:
I'm a little bit concerned about this code getting ignored the next time a new attribute is added to modules. Do you think it would be an improvement to put a list of attributes somewhere like `ModuleBase` and then loop through it here? I suppose it would also need to store which attributes are primitives and which need a `deepcopy`.

Reply:
Yeah, I'm pretty sure this exact scenario is one of our interview questions :laughing: 
Let me think about this a bit more and see how I can improve it. I'm a bit wary of introducing more stuff into `ModuleBase` especially for such an esoteric feature, so I'm going to try and let this piece of code figure it out by itself

Reply:
Cool, agreed that there's a downside to putting this in `ModuleBase`. I wonder if there's a way you could write a test to check this - like, make this section into a helper function and then write a test that calls this function and checks the to_json matches for the new and old modules (excepting things like unique_id, I haven't thought this through).